src/main/java/appeng/container/slot/OptionalSlotFakeFluid.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * This file is part of Applied Energistics 2.
+ * Copyright (c) 2013 - 2018, AlgorithmX2, All rights reserved.
+ *
+ * Applied Energistics 2 is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Applied Energistics 2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Applied Energistics 2.  If not, see <http://www.gnu.org/licenses/lgpl>.
+ */
+
+package appeng.container.slot;
+
+
+import javax.annotation.Nonnull;
+
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.items.IItemHandler;
+
+
+/**
+ * @author BrockWS
+ * @version rv6 - 1/05/2018
+ * @since rv6 1/05/2018
+ */
+public class OptionalSlotFakeFluid extends SlotFakeFluid implements IOptionalSlot
+{
+
+	private final int srcX;
+	private final int srcY;
+	private final int groupNum;
+	private final IOptionalSlotHost host;
+	private boolean renderDisabled = true;
+
+	public OptionalSlotFakeFluid( final IItemHandler inv, final IOptionalSlotHost containerBus, final int idx, final int x, final int y, final int offX, final int offY, final int groupNum )
+	{
+		super( inv, idx, x + offX * 18, y + offY * 18 );
+		this.srcX = x;
+		this.srcY = y;
+		this.groupNum = groupNum;
+		this.host = containerBus;
+	}
+
+	@Override
+	@Nonnull
+	public ItemStack getStack()
+	{
+		if( !this.isSlotEnabled() )
+		{
+			if( !this.getDisplayStack().isEmpty() )
+			{
+				this.clearStack();
+			}
+		}
+
+		return super.getStack();
+	}
+
+	@Override
+	public boolean isSlotEnabled()
+	{
+		if( this.host == null )
+		{
+			return false;
+		}
+
+		return this.host.isSlotEnabled( this.groupNum );
+	}
+
+	public boolean isRenderDisabled()
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add the "@Override" annotation above this method signature [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1161)